src/Core/Service/ServiceException.php

```diff
@@ -124,12 +124,12 @@ public static function requestTransportError(?\Throwable $previous = null): self
         );
     }
 
-    public static function missingAppVersionInfo(): self
+    public static function missingAppVersionInfo(string $missingField): self
     {
         return new self(
             Response::HTTP_BAD_REQUEST,
             self::SERVICE_MISSING_APP_VERSION_INFO,
-            'Error downloading app. The version information was missing.'
+            \sprintf('Error downloading app. The version information was missing: %s', $missingField)
```

Comment:
please use the built-in message pattern with parameters. `{{ stuff }}` placeholder and so on